sklearn/model_selection/_search.py

```diff
@@ -14,30 +14,35 @@
 from collections import Mapping, namedtuple, Sized
 from functools import partial, reduce
 from itertools import product
+import numbers
 import operator
+import time
 import warnings
 
 import numpy as np
 
 from ..base import BaseEstimator, is_classifier, clone
 from ..base import MetaEstimatorMixin, ChangedBehaviorWarning
-from ._split import check_cv
-from ._validation import _fit_and_score
-from ..externals.joblib import Parallel, delayed
 from ..externals import six
+from ..externals.joblib import Parallel, delayed
+from ..metrics.scorer import check_scoring
 from ..utils import check_random_state
 from ..utils.fixes import sp_version
+from ..utils.metaestimators import if_delegate_has_method
 from ..utils.random import sample_without_replacement
 from ..utils.validation import _num_samples, indexable
-from ..utils.metaestimators import if_delegate_has_method
-from ..metrics.scorer import check_scoring
+from ._split import check_cv
+from ._validation import _fit_and_score
+from ..exceptions import FitFailedWarning
 
 
-__all__ = ['GridSearchCV', 'ParameterGrid', 'fit_grid_point',
-           'ParameterSampler', 'RandomizedSearchCV']
+__all__ = ['GridSearchCV', 'UnsupervisedGridSearch', 'ParameterGrid',
+           'fit_grid_point', 'ParameterSampler', 'RandomizedSearchCV',
+           'UnsupervisedRandomizedSearch']
 
 
 class ParameterGrid(object):
+
```

Comment:
Why the extra line? (I feel this is the change of autopep8? ;) If so, I think, we generally don't recommend autopep8)